.github/services/tikv/tikv_tls/action.yml

```diff
@@ -0,0 +1,77 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: tikv_with_tls
+description: 'Behavior test for TiKV with tls'
+
+runs:
+  using: "composite"
+  steps:
+    - name: Copy TiKV Certificate Files
+      shell: bash
+      working-directory: fixtures/tikv
+      run: |
+        mkdir -p /tmp/tikv/ssl
+        cp -r `pwd`/ssl/* /tmp/tikv/ssl
+
+    - name: install tiup
+      shell: bash
+      run: curl --proto '=https' --tlsv1.2 -sSf https://tiup-mirrors.pingcap.com/install.sh | sh
+
+    - name: Start tikv-tls
```

Comment:
Is it a good idea to use docker-compose instead of tiup?

For example: https://github.com/pingcap/tidb-docker-compose/blob/master/docker-compose.yml

Reply:
I tried it, and it seems that `docker-compose` only works properly on linux due to the limitations of `docker-desktop`.😔

> 1. The docker (Linux) bridge network is not reachable from the macOS/Windows host.
> 2. The host networking driver only works on Linux hosts, and is not supported on Docker Desktop for Mac, Docker Desktop for Windows, or Docker EE for Windows Server.

Reply:
> I tried it, and it seems that `docker-compose` only works properly on linux due to the limitations of `docker-desktop`.😔

Emmm, does tiup work correctly on macos?

Reply:
> > I tried it, and it seems that `docker-compose` only works properly on linux due to the limitations of `docker-desktop`.😔
> 
> Does tiup work correctly on macos?

Yes, tiup works fine on macos.

Reply:
I really don't want to depend on tiup anymore.

Reply:
> I really don't want to depend on tiup anymore.

I think the crux of the matter is how to specify the [advertise-client-urls](https://docs.pingcap.com/tidb/stable/pd-configuration-file#advertise-client-urls) parameter, due to the limitations of `docker-desktop-macos/windows`, it's not possible to access containers using the host network mode or directly by container name or fixed ip. Do you have any suggestions?

```plaintext
ConfigInvalid (persistent) at create_dir, context: { service: tikv, service: tikv, path: ef4706da-0389-44da-a31b-4bc02cb79fc7/ } => invalid configuration, source: [/Users/gxd/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/tikv-client-0.2.0/src/pd/cluster.rs:279]: failed to connect to [Member { name: "pd0", member_id: 16382016976602811638, peer_urls: ["http://pd0:2380"], client_urls: ["http://pd0:2379"], leader_priority: 0, deploy_path: "", binary_version: "", git_hash: "", dc_location: "" }]
```